src/qt/ee_debugwindow.cpp

```diff
@@ -0,0 +1,1231 @@
+    {
+        table->setItem(i,0,new QTableWidgetItem("uninitialized"));
+        table->setRowHeight(i, 20);
+    }
+}
+
+// 128-bit register value to hex
+static QString reg_to_string(uint128_t value)
+{
+    char cstr[128];
+    sprintf(cstr, "%08X %08X %08X %08X", value._u32[3], value._u32[2], value._u32[1], value._u32[0]);
+    return QString(cstr);
+}
+
+static QString reg_to_string(uint64_t value)
+{
+    char cstr[128];
+    sprintf(cstr, "%08X %08X", (uint32_t)(value >> 32), (uint32_t)(value & 0xffffffff));
+    return QString(cstr);
+}
+
+static QString reg_to_string(uint32_t value)
+{
+    char cstr[128];
+    sprintf(cstr, "%08X", value);
+    return QString(cstr);
+}
+
+static QString reg_to_string(float value)
+{
+    char cstr[128];
+    sprintf(cstr, "%g", value);
+    return QString(cstr);
+}
+
+// word as hex, ascii, floating point
+static QString memory_data_to_string(uint32_t data)
+{
+    char* charp = (char*)(&data);
+
+    char ascii[5];
+
+    for (int j = 0; j < 4; j++)
+    {
+        if(charp[j] >= ' ' && charp[j] <= '~')
+            ascii[j] = charp[j];
+        else
+            ascii[j] = '.';
+    }
+    ascii[4] = '\0';
+    char fp[64];
+    sprintf(fp, "%g", *(float*)(&data));
+    return QString(reg_to_string(data) + "    " + QString(ascii) + "    " + QString(fp));
+}
+
+static uint32_t hex_address_to_u32(QString str)
+{
+    char* charp = str.toUtf8().data();
+    return strtoul(charp, nullptr, 16);
+}
+
+
+void EEDebugWindow::set_breakpoint_list(EEBreakpointList *list)
+{
+    ee_breakpoint_list = list;
+}
+
+void EEDebugWindow::set_breakpoint_list(IOPBreakpointList *list)
+{
+    iop_breakpoint_list = list;
+}
+
+void EEDebugWindow::update_break_continue_button()
+{
+    if(emulation_running())
+        break_continue_button->setText("Break");
+    else
+        break_continue_button->setText("Continue");
+}
+
+void EEDebugWindow::setup_ui_formatting()
+{
+    setup_qtable(disassembly_table, EE_DEBUGGER_DISASSEMBLY_SIZE);
+    setup_qtable(breakpoint_table, EE_DEBUGGER_MAX_BREAKPOINTS);
+    setup_qtable(memory_table, EE_DEBUGGER_MEMORY_SIZE);
+    setup_qtable(register_table, ee_registers.size() + 1); // +1 for the PC
+    setup_qtable(stack_table, EE_DEBUGGER_STACK_SIZE);
+
+    add_breakpoint_combo->addItem("Add Breakpoint...");
+    for(auto name : breakpoint_names)
+    {
+        add_breakpoint_combo->addItem(QString(name));
+    }
+
+    cpu_combo->addItem("EE");
+    cpu_combo->addItem("IOP");
+}
+
+void EEDebugWindow::update_ui()
+{
+    update_break_continue_button();
+    update_registers();
+    update_disassembly();
+    update_memory();
+    update_stack();
+    update_breakpoints();
+}
+
+void EEDebugWindow::update_registers()
+{
+    uint32_t r = 0;
+    auto set_next = [&r, this](QString q){register_table->item(r++,0)->setText(q);};
+
+    if(current_cpu == DebuggerCpu::EE)
+    {
+        // program counter
+        set_next(QString("pc  ") + reg_to_string(ee->get_PC()));
```

Comment:
Instead of using concatenation here, QString::arg can be used instead

```suggestion
        set_next(QString("pc  %1").arg(reg_to_string(ee->get_PC())));
```